src/Merchants/Content/Merchant/Api/OrderController.php

```diff
@@ -0,0 +1,146 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Production\Merchants\Content\Merchant\Api;
+
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Core\System\StateMachine\StateMachineRegistry;
+use Shopware\Core\System\StateMachine\Transition;
+use Shopware\Production\Merchants\Content\Merchant\Exception\OrderAlreadyCompletedException;
+use Shopware\Production\Merchants\Content\Merchant\MerchantEntity;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"merchant-api"})
+ */
+class OrderController
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $merchantRepository;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var StateMachineRegistry
+     */
+    private $stateMachineRegistry;
+
+    public function __construct(
+        EntityRepositoryInterface $merchantRepository,
+        EntityRepositoryInterface $orderRepository,
+        StateMachineRegistry $stateMachineRegistry
+    ) {
+        $this->merchantRepository = $merchantRepository;
+        $this->orderRepository = $orderRepository;
+        $this->stateMachineRegistry = $stateMachineRegistry;
+    }
+
+    /**
+     * @Route(name="merchant-api.orders.load", path="/merchant-api/v{version}/orders", methods={"GET"})
+     */
+    public function load(MerchantEntity $merchant, Request $request): JsonResponse
+    {
+        $orderState = $request->query->getAlnum('state');
+        $limit = $request->query->getInt('limit', 25);
+        $offset = $request->query->getInt('offset', 0);
+
+        $criteria = new Criteria();
+        $criteria->addAssociation('merchants');
+        $criteria->addAssociation('deliveries');
+        $criteria->addAssociation('lineItems');
+        $criteria->setTotalCountMode(Criteria::TOTAL_COUNT_MODE_EXACT)
+            ->addFilter(new EqualsFilter('merchants.id', $merchant->getId()))
+            ->setLimit($limit)
+            ->setOffset($offset);
+
+        if ($orderState !== '') {
+            $criteria->addFilter(
+                new EqualsFilter('stateMachineState.technicalName', $orderState)
+            );
+        }
+
+        $orders = $this->orderRepository->search($criteria, Context::createDefaultContext());
+
+        if ($orders->count() <= 0) {
+            throw new NotFoundHttpException('No orders found.');
```

Comment:
Why not an empty list?

Reply:
Because we throw this exception at other api points. Consitent Behaviour.

Reply:
I guess for the angular guys it's easier to have an empty result instead of additional handling of status code.
@FloBWer

Reply:
Yes thats the eseaest way for us. We have an empty state in our list

Reply:
Done :+1: